standardtracer/propagation.go

```diff
@@ -44,9 +45,11 @@ func (p *splitTextPropagator) InjectSpan(
 		fieldNameSampled: strconv.FormatBool(sc.raw.Sampled),
 	}
 	sc.Lock()
-	splitTextCarrier.TraceAttributes = make(map[string]string, len(sc.raw.Attributes))
-	for k, v := range sc.raw.Attributes {
-		splitTextCarrier.TraceAttributes[k] = v
+	if l := len(sc.raw.Attributes); l > 0 {
+		splitTextCarrier.TraceAttributes = make(map[string]string, l)
```

Comment:
I was thinking of this scenario:
```go
carrier := &SplitTextCarrier{TracerState: request.Header, TraceAttributes: requestHeader}
injector.inject(span, carrier)
```
I.e. it's not the job of the injector to allocate memory for target maps, and in this case I am not allocating any memory at all, only using already available `request.Header` map.